### PR TITLE
refine tensorflow support doc

### PR DESCRIPTION
**Pull request by yangw1234**
## What changes were proposed in this pull request?

add tensorflow saver test and refine tensorflow support doc

## How was this patch tested?

added unit tests

### CODE REVIEW

**Comment by jason-dai** [comment]
Need to add how to export Tensoflow model

**Comment by yangw1234** [comment]
added tensorflow saver python api and docs

**Comment by yangw1234** [comment]
pr validation pass

**Comment by yiheng** [comment]
LGTM

File: docs/docs/ProgrammingGuide/tensorflow-support.md [hunk 1]
```diff
@@ -1,16 +1,96 @@
 ## **Loading a tensorflow model from tensorflow model file**
 
 If you have a pre-trained tensorflow model saved in a ".pb" file. You load it
```
Comment (jason-dai) [inline docs/docs/ProgrammingGuide/tensorflow-support.md]: You can load it

File: docs/docs/ProgrammingGuide/tensorflow-support.md [hunk 2]
```diff
@@ -1,16 +1,96 @@
...
-into BigDL using `Model.load_tensorflow` api.
+into BigDL using `Model.load_tensorflow` api. 
 
-**Python:**
+For more information on how to generate
+the ".pb" file, you can refer to [A Tool Developer's Guide to TensorFlow Model Files](https://www.tensorflow.org/extend/tool_developers/).
+Specifically, you should generate a model definition file and a set of checkpoints, then use the [freeze_graph](https://github.com/tensorflow/tensorflow/blob/v1.0.0/tensorflow/python/tools/freeze_graph.py)
+script to freeze the graph definition and weights in checkpoints into a single file.
+
+**Generate model definition file and checkpoints in python**
```
Comment (jason-dai) [inline docs/docs/ProgrammingGuide/tensorflow-support.md]: Generate model definition file and checkpoints in TensorFlow

File: docs/docs/ProgrammingGuide/tensorflow-support.md [hunk 3]
```diff
@@ -1,16 +1,96 @@
...
+    tf.train.write_graph(sess.graph, '/tmp/model', 'test.pbtxt')
+```
+
+**Freeze graph definition and checkpoints into a single ".pb" file in shell**
+```shell
+wget https://raw.githubusercontent.com/tensorflow/tensorflow/v1.0.0/tensorflow/python/tools/freeze_graph.py
+python freeze_graph.py --input_graph /tmp/model/test.pbtxt --input_checkpoint /tmp/model/test.chkp --output_node_names=output --output_graph "/tmp/model/test.pb"
+```
+
+**Load tensorflow model in Scala**
```
Comment (jason-dai) [inline docs/docs/ProgrammingGuide/tensorflow-support.md]: Load TensorFlow model in BigDL

File: docs/docs/ProgrammingGuide/tensorflow-support.md [hunk 4]
```diff
@@ -1,16 +1,96 @@
...
+```scala
+import com.intel.analytics.bigdl.utils._
+val path = "/tmp/model/test.pb"
+val inputs = Seq("Placeholder")
+val outputs = Seq("output")
+val model = TensorflowLoader.load(path, Seq("Placeholder"), Seq("output"), ByteOrder.LITTLE_ENDIAN)
```
Comment (jason-dai) [inline docs/docs/ProgrammingGuide/tensorflow-support.md]: val model = Module.loadTF(...)

File: docs/docs/ProgrammingGuide/tensorflow-support.md [hunk 5]
```diff
@@ -1,16 +1,96 @@
...
+```
+
+**Load tensorflow model in python**
```
Comment (jason-dai) [inline docs/docs/ProgrammingGuide/tensorflow-support.md]: Delete this line

Comment (yangw1234) [inline docs/docs/ProgrammingGuide/tensorflow-support.md]: create another pr #1253 to fix this

**Pull request merged**